From 1da9df7f4ea442863cf4af24e3cba29d79de5ea3 Mon Sep 17 00:00:00 2001
From: Sheng Long Wang <shenglong.wang.ext@siemens.com>
Date: Fri, 13 Dec 2019 12:35:56 +0800
Subject: [PATCH 13/22] fix(4169461):fixed eth link down when autoneg off

1.dp83867: enable robust auto-mdix

CFG3[9] Robust Auto-MDIX option allows significantly improve link
detection in case dp83867 is configured in manual mode and reduce
link detection time.
DM says: "If link partners are configured to operational modes that
are not supported by normal Auto MDI/MDIX mode (like Auto-Neg versus
Force 100Base-TX or Force 100Base-TX versus Force 100Base-TX), this
Robust Auto MDI/MDIX mode allows MDI/MDIX resolution and prevents
deadlock."

Hence, enable this option by default as there are no known reasons not
to do so.

Signed-off-by: Sheng Long Wang <shenglong.wang.ext@siemens.com>
---
 drivers/net/phy/dp83867.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/net/phy/dp83867.c b/drivers/net/phy/dp83867.c
index 88cd51af334c..68d7667f1297 100644
--- a/drivers/net/phy/dp83867.c
+++ b/drivers/net/phy/dp83867.c
@@ -319,12 +319,13 @@ static int dp83867_config_init(struct phy_device *phydev)
 		}
 	}
 
+	val = phy_read(phydev, DP83867_CFG3);
+
 	/* Enable Interrupt output INT_OE in CFG3 register */
-	if (phy_interrupt_is_valid(phydev)) {
-		val = phy_read(phydev, DP83867_CFG3);
+	if (phy_interrupt_is_valid(phydev))
 		val |= BIT(7);
-		phy_write(phydev, DP83867_CFG3, val);
-	}
+	val |= BIT(9);
+	phy_write(phydev, DP83867_CFG3, val);
 
 	if (dp83867->port_mirroring != DP83867_PORT_MIRROING_KEEP)
 		dp83867_config_port_mirroring(phydev);
-- 
2.17.1

